packages/java-edition/src/mcfunction/parser/argument.ts

```diff
@@ -1321,14 +1330,27 @@ function selector(): core.Parser<EntitySelectorNode> {
 				},
 			},
 		]),
-		(res) => {
+		(res, _, ctx) => {
+			const selector = res.children.find(core.LiteralNode.is)!.value.slice(1)
+			const release = ctx.project['loadedVersion'] as
+				| ReleaseVersion
+				| undefined
+			// Invalid @ selector (including @n before 1.21)
+			if (
+				!(EntitySelectorVariable.is(selector)) ||
+				(release && ReleaseVersion.cmp(release, '1.21') < 0 &&
+					selector === 'n')
```

Comment:
This should re-use the same function as the completer to determine the currently allowed selector variables, although this whole logic should be moved to the `selectorPrefix` function (as explained in my later comment)